core/modules/replica/SqlRequest.cc

```diff
@@ -292,39 +251,652 @@ void SqlRequest::_analyze(bool success,
+void SqlQueryRequest::notify(util::Lock const& lock) {
+
+    LOGS(_log, LOG_LVL_DEBUG, context() << __func__ <<
+        "[" << ProtocolRequestSql_Type_Name(requestBody.type()) << "]");
+
+    notifyDefaultImpl<SqlQueryRequest>(lock, _onFinish);
+}
+
+
+SqlCreateDbRequest::Ptr SqlCreateDbRequest::create(
+        ServiceProvider::Ptr const& serviceProvider,
+        boost::asio::io_service& io_service,
+        string const& worker,
+        std::string const& database,
+        CallbackType const& onFinish,
+        int priority,
+        bool keepTracking,
+        shared_ptr<Messenger> const& messenger) {
+
+    return Ptr(new SqlCreateDbRequest(
+        serviceProvider,
+        io_service,
+        worker,
+        database,
+        onFinish,
+        priority,
+        keepTracking,
+        messenger
+    ));
+}
+
+
+SqlCreateDbRequest::SqlCreateDbRequest(
+        ServiceProvider::Ptr const& serviceProvider,
+        boost::asio::io_service& io_service,
+        string const& worker,
+        std::string const& database,
+        CallbackType const& onFinish,
+        int priority,
+        bool keepTracking,
+        shared_ptr<Messenger> const& messenger)
+    :   SqlBaseRequest(serviceProvider,
+                       io_service,
+                       worker,
+                       0 /* maxRows */,
+                       priority,
+                       keepTracking,
+                       messenger),
+        _onFinish(onFinish) {
+
+    // Finish initializing the request body's content
+    requestBody.set_type(ProtocolRequestSql::CREATE_DATABASE);
+    requestBody.set_database(database);
+}
+
+
+void SqlCreateDbRequest::notify(util::Lock const& lock) {
+
+    LOGS(_log, LOG_LVL_DEBUG, context() << __func__ <<
+        "[" << ProtocolRequestSql_Type_Name(requestBody.type()) << "]");
+
+    notifyDefaultImpl<SqlCreateDbRequest>(lock, _onFinish);
+}
+
+
+SqlDeleteDbRequest::Ptr SqlDeleteDbRequest::create(
+        ServiceProvider::Ptr const& serviceProvider,
+        boost::asio::io_service& io_service,
+        string const& worker,
+        std::string const& database,
+        CallbackType const& onFinish,
+        int priority,
+        bool keepTracking,
+        shared_ptr<Messenger> const& messenger) {
+
+    return Ptr(new SqlDeleteDbRequest(
+        serviceProvider,
+        io_service,
+        worker,
+        database,
+        onFinish,
+        priority,
+        keepTracking,
+        messenger
+    ));
+}
+
+
+SqlDeleteDbRequest::SqlDeleteDbRequest(
+        ServiceProvider::Ptr const& serviceProvider,
+        boost::asio::io_service& io_service,
+        string const& worker,
+        std::string const& database,
+        CallbackType const& onFinish,
+        int priority,
+        bool keepTracking,
+        shared_ptr<Messenger> const& messenger)
+    :   SqlBaseRequest(serviceProvider,
+                       io_service,
+                       worker,
+                       0 /* maxRows */,
```

Comment:
nit: IMO `// maxRows` is a little better, safer (b/c `/* ... */` doesn't nest, and because there's nothing else on this line `//` is preferable.

Reply:
(and so on for the other cases in this file)